python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -60,12 +60,35 @@
 from .system import StepExecutionContext
 
 
+class ExecutionProperties(
+    NamedTuple(
+        "_ExecutionProperties",
+        [
+            ("step_description", PublicAttr[str]),
+            ("op_execution_context", PublicAttr["OpExecutionContext"]),
```

Comment:
having to pass the OpExecutionContext here is kind of annoying, but the alternative is doing `isinstance(context, AssetExecutionContext)` checks in the core execution code so that we can access `context.op_execution_context` when the context is an `AssetExecutionContext`

Reply:
another option could be to move the functions needed from `OpExecutionContext` in the core execution path into the `ExecutionProperties` named tuple. I can look into that option as well